docs/ru/http-headers.md

```diff
@@ -0,0 +1,193 @@
+
+$etag = (new ETag())->withTag('56d-9989200-1132c580', true)->inject($response);
+```
+
+## Заголовки кеширования
+
+[RFC7234](https://tools.ietf.org/html/rfc7234)
+
+### Age
+
+В заголовке `Age` задаётся количество секунд с момента модификации ресурса. Поэтому все значения, имеющие символы,
+отличные от десятичных цифр, будут помечены как невалидные.
+
+### Expires
+
+Дата истечения срока актуальности сущности. Класс заголовка `Expires` наследуется от класса `Date`.
+
+```php
+use Yiisoft\Http\Header\Value\Cache\Expires;
+/** @var \Psr\Http\Message\ResponseInterface $response */
+
+$dateHeader = (new Expires(new DateTimeImmutable('+1 day')))
+    ->inject($response);
+```
+
+### Warning
+
+Заголовок для дополнительной информации об ошибках.
+```php
+use \Yiisoft\Http\Header\Value\Cache\Warning;
+/** @var \Psr\Http\Message\ResponseInterface $response */
+
+$dateHeader = Warning::createHeader()
+    ->withValue((new Warning())->withDataset(Warning::RESPONSE_IS_STALE, '-', 'Response is stale'))
+    ->withValue((new Warning())->withDataset(Warning::REVALIDATION_FAILED, '-', 'Revalidation failed'))
+    ->inject($response);
+```
+
+### Cache-Control
+
+Заголовок `Cache-Control` задаёт правила кеширования. Его особенность заключается в том, что вместо значений и их
+параметров используются директивы. С целью переиспользования кода парсера директива без аргумента парсится как значение,
+а директива с аргументом — как параметр без значения. Однако для пользователя выделен отдельный метод для записи
+директивы и аргумента `withDirective(string $directive, string $argument = null)`
+
+```php
+$header = \Yiisoft\Http\Header\Value\Cache\CacheControl::createHeader()
+    ->withDirective('max-age', '27000')
+    ->withDirective('no-transform')
+    ->withDirective('foo', 'bar');
+```
+
+Для всех стандартных директив, которые описаны в [главе 5.2 RFC7234](https://tools.ietf.org/html/rfc7234#section-5.2),
+аргументы будут проходить этап валидации. Если вы указываете директиву вручную, будьте готовы к тому, что метод
+`withDirective()` может выбросить исключение.
+
+```php
+use Yiisoft\Http\Header\Value\Cache\CacheControl;
+
+// исключения будут брошены в каждом из этих случаев:
+(new CacheControl())->withDirective('max-stale'); // у директивы max-stale должен быть аргумент
+(new CacheControl())->withDirective('max-age', 'not numeric'); // аргумент директивы max-age должен быть числовым
+(new CacheControl())->withDirective('max-age', '-456'); // допускаются только цифры
+(new CacheControl())->withDirective('private', 'ETag,'); // нарушение синтаксиса списка заголовков
+(new CacheControl())->withDirective('no-store', 'yes'); // директива no-store не принимает аргумент
+
+// Исключение выброшено не будет, однако все элементы коллекции будут невалидными
+CacheControl::createHeader()->withValue('max-stale, max-age=test, private="ETag,", no-store=yes');
+```
+
+### Pragma
+
+Устаревший заголовок из HTTP/1.0, использующийся для обратной совместимости.
```

Comment:
Do we need to implement depreated things at all?

Reply:
It is a part of standart. More Web Clients works only with this cache header